### ISSUE TITLE
Tag Maker with Non English Chars

### ISSUE THREAD

**Issue by fjunior87** [Adobe-Consulting-Services/acs-aem-tools#122]
Hi,

I am trying to use the ACS Tools Tag Maker in order to create a tagging structure but I'm facing some issues because I have tags with non English chars like: Café,Açaí.

I can make the tag be imported correctly by using the notation: Café{{cafe}}, but the title appears broken in the special char.

Does someone already faced this kind of issue? Is there any workaround?

This Question was also made on Adobe [Forum](http://help-forums.adobe.com/content/adobeforums/en/experience-manager-forum/adobe-experience-manager.topic.5_ciot.html/forum__llek-hi_i_am_tryingt.html), any tips are welcome.

**Comment by davidjgonzalez** [comment]
Are you sure the source Csv is Utf-8 encoded? The tag maker form allows you to specify the char encoding but defaults to Utf-8

**Comment by fjunior87** [comment]
Hi @davidjgonzalez, Yes I am.

I've saved the csv from a Excel Spreadsheet then I opened it on Notepad++ and changed the encoding to UTF-8.

Is there anything else I could do?

Thanks

**Comment by davidjgonzalez** [comment]
Can you attach the Csv (or a few rows w offending entries) so we can take a look at it? 

As long as you the Csv attached exhibits he problem that should be sufficient.

**Comment by fjunior87** [comment]
Follow the csv. I renamed it to xls to be able to upload it.
[Planilha sem título - Página1.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-tools/files/73498/Planilha.sem.titulo.-.Pagina1.xlsx)

**Comment by davidjgonzalez** [comment]
This works for me using your file (renamed to .csv) .. AEM 6.0 SP2..

![2015-12-28 at 4 59 pm](https://cloud.githubusercontent.com/assets/1451868/12026154/5c463fa0-ad84-11e5-8212-4c4c7e9e086d.png)

![2015-12-28 at 4 58 pm](https://cloud.githubusercontent.com/assets/1451868/12026148/42ea2bde-ad84-11e5-889a-1608ed29cc95.png)

**Comment by fjunior87** [comment]
Hi @davidjgonzalez, I am using AEM 6.1 on Windows, would this impact?

**Comment by davidjgonzalez** [comment]
I'll try to find time to test on 6.1 might be a few days. Can you confirm this doesn't work on 6.0?

Windows could be the issue.. Seems strange if the file you uploaded for me doesn't work for you though.. I'd thibk the problem would be in the saving of the file and not the uploading.

**Comment by fjunior87** [comment]
Hi @davidjgonzalez I tried with 6.0 and the same happened. Maybe it is an issue with Windows.

**Comment by davidjgonzalez** [comment]
Did you try w the exact file you attached to the GH issue?  If that file is untouched after saving and uploading, I'm not sure how it could be a Windows issue (unless Windows is doing sormthing really strange to the file on save/upload)

**Comment by fjunior87** [comment]
Hi,

It worked using the Cp1252 encoding.

I've tested creating another csv from an Excel Spreadsheet. And it worked.

Thanks for all help.

**Comment by davidjgonzalez** [comment]
@fjunior87 did you just set the char encodingin the TagMaker form to CP1252?

**Comment by fjunior87** [comment]
@davidjgonzalez Yes. I've created another csv from an Excel Spreadsheet. Then in the form I used CP1252 and those words with those special chars were inserted correctly.

Follow the CSV.
[tag.txt](https://github.com/Adobe-Consulting-Services/acs-aem-tools/files/81111/tag.txt)

Please rename it to csv and the separator is ";"

**Comment by fjunior87** [comment]
Hi @davidjgonzalez, I was thinking in add a code to handle those special chars so we don't need to use the notation {{}}. My idea was to use the Normalizer class in the LowercaseWithDashesConverterImpl. What do you think? Does it make sense? or it would be better create a new converter?

My idea is:
If we have a string like: Avião. It would create a tag with title Avião and name aviao.

Where do you think would be better to put this code?

**Comment by davidjgonzalez** [comment]
@fjunior87 mm - this makes sense to me, but the flip-side is i've only ever worked where the primary/core language was English. On non-English apps, is there ever a good reason to use the character variants? 

@joerghoh you have a non-English perspective on this?

**Comment by davidjgonzalez** [comment]
@fjunior87 thinking on this a little more; i think we should make a a new converter for this use case since other folks may expect/want/depend on the original implementation. You probably could just extend LowercaseWithDashesConverterImpl and reset the values on the resulting TagData obj as needed.
